src/oss/confluent.sh

```diff
@@ -1308,6 +1310,60 @@ extract_json_config() {
     _retval="${parsed_json}"
 }
 
+produce_command() {
+    local topicname="${1}"
+    local args="$@"
+
+    if [[ "x${topicname}" == "x" ]]; then
```

Comment:
With double-brackets you dont need to quote, nor check for x-style empty, this is sufficient:
`if [[ -z $topicname ]]; then .. # topic was empty`

Reply:
@edenhill , I was following the convention that @kkonstantine requested.  But I may have misunderstood what he intended, so let's have him add a comment here.  I agree that your syntax is much cleaner

Reply:
@edenhill is right in general, but we can keep a consistent, fail-safe, style and quote as well.

Reply:
For instance, not quoting in an `if` block and quoting in a variable assignment, might make you miss quoting somewhere that is essential. That's why I've followed a 'quote-by-default' pattern almost everywhere. There are cases where quoting might not produce the desired result (don't have an example handy).

Reply:
@kkonstantine , so this means no change required to current code, correct?